osbs/utils/__init__.py

```diff
@@ -726,3 +726,36 @@ def copy(self):
             namespace=self.namespace,
             repo=self.repo,
             tag=self.tag)
+
+
+class UserWarningsStore(object):
+    def __init__(self):
+        # asctime (platform:arch)? - name - levelname - message
+        self.regex = r'^.+ - USER_WARNING - (\{.*\})$'
+        self.__user_warnings = set()
+
+    def is_user_warning(self, line):
+        return re.match(self.regex, line)
+
+    def store(self, line):
+        """
+        Extract data from given log record with USER_WARNING level
+        and store an understandable message in set
+        """
+        data_search = re.search(self.regex, line)
+        if not data_search:
+            return
+
+        try:
+            data = json.loads(data_search.group(1))
+        except ValueError:
+            message = 'Incorrect JSON data input for a user warning: %s'
+            logger.warning(message, data_search.group(1))
+            return
+
+        message = data['message']
+
+        self.__user_warnings.add(message)
```

Comment:
just use directly data

self._user_warnings.add(data['message'])

Reply:
I've made it for future extensions, maybe we'll have to combine retrieved data, so I've decided to save the message into a corresponding variable